src/components/AddItem.jsx

```diff
@@ -1,7 +1,102 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { collection, addDoc } from 'firebase/firestore';
+import { db } from '../lib/firebase';
+
+const radioButtonOptions = [
+  {
+    option: 'soon',
+    value: 7,
+    defaultChecked: true,
+  },
+  {
+    option: 'kind of soon',
+    value: 14,
+    defaultChecked: false,
+  },
+  {
+    option: 'not soon',
+    value: 30,
+    defaultChecked: false,
+  },
+];
 
 function AddItem() {
-  return <div>Add Item</div>;
+  const [itemName, setItemName] = useState('');
+  const [frequency, setFrequency] = useState(7);
+  const [notification, setNotification] = useState('');
+
+  //retrive the token from localStorage
+  const token = localStorage.getItem('list-token');
+
+  //handle submit for the form
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    //create new doc reference for text input and save to firestore database
+    //use a token to save new items into specific collection (shopping list)
+    try {
+      const docRef = await addDoc(collection(db, token), {
+        //data points being sent to firebase, object format
+        //should we convert frequency to a number? or send in as a string?
```

Comment:
I think Number is the right way to go. When we get to Issue 10 and estimating the next purchase dates, there's a comment about data inputs. I poked around the function a little and all I see are numbers in the tests.

I really appreciate your comments and how easy it was to follow everything you guys did 👍

Reply:
Seconded what Shelley said about comments. The 'number or string?' comment is great to show current thinking and possible improvement to whoever looks at code now, or 6 months from now, or 6 years (including yourself), especially when juggling lots of tasks in same time. It's also best way to avoid this scenario: https://the-collab-lab.slack.com/archives/CLRUG5G8G/p1642473077036000